ocean_lib/ocean/ocean_assets.py

```diff
@@ -666,6 +664,111 @@ def pay_for_service(
 
         return tx_id
 
+    @enforce_types
+    def pay_for_compute_service(
+        self,
+        datasets: List[ComputeInput],
+        algorithm_data: Union[ComputeInput, AlgorithmMetadata],
+        compute_environment: str,
+        valid_until: int,
+        consume_market_order_fee_address: str,
+        wallet: Wallet,
+        consumer_address: Optional[str] = None,
+    ):
+        data_provider = DataServiceProvider
+
+        if not consumer_address:
+            consumer_address = wallet.address
+
+        initialize_response = data_provider.initialize_compute(
+            [x.as_dictionary() for x in datasets],
+            algorithm_data.as_dictionary(),
+            datasets[0].service.service_endpoint,
+            consumer_address,
+            compute_environment,
+            valid_until,
+        )
+
+        result = initialize_response.json()
+        for i, item in enumerate(result["datasets"]):
+            self._start_or_reuse_order_based_on_initialize_response(
+                datasets[i],
+                item,
+                consume_market_order_fee_address,
+                datasets[i].consume_market_order_fee_token,
+                datasets[i].consume_market_order_fee_amount,
+                wallet,
+                consumer_address,
+            )
+
+        if "algorithm" in result:
+            self._start_or_reuse_order_based_on_initialize_response(
+                algorithm_data,
+                result["algorithm"],
+                consume_market_order_fee_address,
+                datasets[i].consume_market_order_fee_token,
+                datasets[i].consume_market_order_fee_amount,
```

Comment:
Why would you grab the dataset consume_market_order_fee_token and amount for the algorithm?  

I think we need to restructure the ComputeImput so it can either contain an Asset + Service or an AlgorithmMetadata.